enumerable_method.rb

```diff
@@ -29,180 +30,252 @@ def my_select
     newarray
   end
 
-  def my_all?(param = nil)
-    return false if param.class == Regexp
+  def my_count(item = nil)
+    counter = 0
+    if item.nil?
+      return length unless block_given?
 
-    t = true
-    if block_given?
-      return true unless block_given?
+      each do |x|
+        counter += 1 if yield(x)
+      end
+    else
+      my_each do |y|
+        counter += 1 if y == item
+      end
+    end
+    counter
+  end
 
+  def my_map
+    return to_enum(:my_map) unless block_given?
+
+    newarray = []
+    if self.class == Array
       my_each do |x|
-        t = if_nil_false_return_false yield(x)
-        break if t == false
+        newarray << yield(x) ? true : false
+      end
+    else
+      i = first
+      while i <= last
+        newarray << yield(i) unless yield(i).nil?
+        i += 1
       end
-    elsif param.nil?
+    end
+    newarray
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity
+  # rubocop:disable Metrics/MethodLength
+  # rubocop:disable Metrics/PerceivedComplexity
+  def my_all?(param = nil)
+    t = true
+
+    if block_given?
       my_each do |x|
-        t = if_nil_false_return_false x
+        t = yield(x).nil? || yield(x) == false ? false : true
         break if t == false
       end
+    elsif !param.nil?
+      if param.class == Class
+        my_each do |x|
+          return false unless x.is_a?(param)
+        end
+      elsif param.class == Regexp
+        my_each do |x|
+          return false unless x.match(param)
+        end
+      else
+        my_each do |x|
+          return false unless x == param
+        end
+      end
     else
       my_each do |x|
-        t = if_is_a_class(x, param)
-        break if t == false
+        return false unless x
       end
     end
     t
   end
 
   def my_any?(param = nil)
-    return false if param.class == Regexp
     return false if empty?
 
     t = true
```

Comment:
`my_any?` is not working for cases where the argument is not a member of the collection. e.g. `[1,2].my_any?(3)` equals `true` instead of `false`

[Optional]

I think you should initialise `t` as false.

Currently, `my_any?` returns `true` even when the argument is not a member of the collection. See example:

```ruby
irb(main):004:0> [1,2,3].any?(5)
=> false
irb(main):005:0> [1,2,3].my_any?(5)
=> true

```

If you want to do it this way, you would need to change the `unless` to `if` as in `return true if x == param` for case where a `Class` or `RegEx` is not the argument passed.

Reply:
thank you,  you recommendations works for me